pgpainless-core/src/main/java/org/pgpainless/key/info/KeyInfo.java

```diff
@@ -64,4 +81,24 @@ public static String getCurveName(PGPPublicKey publicKey) {
     public static String getCurveName(ECPublicBCPGKey key) {
         return ECUtil.getCurveName(key.getCurveOID());
     }
+
+    /**
+     * Returns indication that a secret key is encrypted.
+     *
+     * @param secretKey A secret key to examine.
+     * @return true if secret key is encrypted, false otherwise.
+     */
+    public static boolean isEncrypted(PGPSecretKey secretKey) {
+        return secretKey.getS2KUsage() != 0 && secretKey.getS2K().getType() != S2K.GNU_DUMMY_S2K;
+    }
```

Comment:
If I get a key:
 - primary key (decrypted)
 - gnu dummy s2k

then:
 - `isFullyDecrypted` should return true.
 - `isFullyEncrypted` should return false

If I get a key:
 - primary key (encrypted)
 - gnu dummy s2k

then:
 - `isFullyDecrypted` should return false.
 - `isFullyEncrypted` should return true

For individual packets, isEncrypted or isDecrypted of a GNU_DUMMY_S2K is not a meaningful question - I would suggest to return null, and handle the case in `isFully...`

Reply:
@tomholub do you mean we should just ignore subkeys with GNU_DUMMY_S2K? what if primary key has dummy S2K?

Reply:
@tomholub so should we just ignore dummy s2k?

Reply:
@tomholub Tried to do that in the 3c32bf6

Reply:
We should not allow s2k to change the output of isFullyEncrypted or isFullyDecrypted. Best of all would be to skip them entirely somehow.